daffodil-tdml-processor/src/test/java/org/apache/daffodil/tdml/TestRunnerFactory.java

```diff
@@ -39,6 +40,7 @@ public void testPrimaryConstructor() {
     Right<scala.xml.Elem, String> rightURI = new Right<>(tdmlUri.toString());
     Runner runner = new Runner(
       rightURI,
+      TDMLImplementation.allValues().head(),
```

Comment:
Here's a case where .head() is still used. Should this be `TDMLImplementation.Daffodil`?

Reply:
Since I will change the parameter's type as well, I will pass Option.apply(null) instead of TDMLImplementation.allValues().head().  I noticed IDEA warns that TestRunnerFactory is calling Runner's private constructor, but sbt still compiles TestRunnerFactory fine, so that's why I think Java can call a Scala private constructor (TestRunnerFactory is written in Java, not Scala).